src/integrations/BingAds/utils.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable camelcase */
+const EXCLUSION_KEYS = [
+  'event',
+  'category',
+  'category_id',
+  'currency',
+  'total',
+  'value',
+  'revenue',
+  'ecomm_category',
+  'transaction_id',
+  'order_id',
+  'checkout_id',
+  'ecomm_pagetype',
+  'pagetype',
+  'query',
+  'products',
+  'product_id',
+  'sku',
+  'event_action',
+];
+
+const DEFAULT_PAGETYPE = 'other';
+
+const buildCommonPayload = (message) => {
+  const { event, properties = {} } = message;
+  const { category, currency, total, value, revenue } = properties;
+  const payload = {
+    event_label: event,
+    event_category: category,
+    currency,
+    revenue_value: total || value || revenue,
+  };
+  return payload;
+};
+
+const handleProductsArray = (properties) => {
+  const productIds = [];
+  const items = [];
+
+  const products = Array.isArray(properties.products) ? properties.products : [properties];
+
+  products.forEach((product) => {
+    const { product_id, sku, price, quantity = 1 } = product;
+    const productId = product_id || sku;
+
+    if (productId) {
+      productIds.push(productId);
+      const item = { id: productId, quantity };
+      if (price && !Number.isNaN(price)) {
+        item.price = price;
+      }
+      items.push(item);
+    }
+  });
+
+  const payload = {};
+  if (items.length > 0) {
+    payload.ecomm_prodid = productIds;
+    payload.items = items;
+  }
+
+  return payload;
+};
+
+const buildEcommPayload = (message) => {
+  const { properties = {} } = message;
+  const {
+    category_id,
+    total,
+    value,
+    ecomm_category,
+    transaction_id,
+    order_id,
+    checkout_id,
+    ecomm_pagetype,
+    pagetype,
+    query,
+  } = properties;
+
+  const ecommPayload = {
+    ecomm_totalvalue: total || value,
+    search_term: query,
+    ecomm_query: query,
+    ecomm_category: ecomm_category || category_id,
+    transaction_id: transaction_id || order_id || checkout_id,
+    ecomm_pagetype: ecomm_pagetype || pagetype || DEFAULT_PAGETYPE,
+  };
+  const payload = handleProductsArray(properties);
+
```

Comment:
don't we need to handle the known scenario where page type is required if prodid is included?

Reply:
missing pagetype check is not required now.